### PR TITLE
Fix(Room): Show better error msg when index is negative in ofind/odel

### LINKED ISSUE #165: Bug(Room): Negative indices in room commands cause a different error message from out of range indices

### PR DESCRIPTION
**Pull request by DrWala**
### What this does:
- Shows a clearer error message when indices are negative vs out of range
- Closes #165 

### How to test:
- Run `oedit` or `odel` with a negative index. Expect to see "Index is not a non-zero unsigned integer." 
- Previously, it would just show the command usage. Ensure that is not the case now.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java [hunk 1]
```diff
@@ -42,7 +41,7 @@ public EditRoomCommand parse(String userInput) throws ParseException {
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditRoomCommand.MESSAGE_USAGE), pe);
+            throw pe;
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java:44]: Maybe just don't catch it?

File: src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java [hunk 2]
```diff
@@ -20,12 +18,12 @@
...
     public DeleteRoomCommand parse(String userInput) throws ParseException {
+        Index index;
         try {
-            Index index = ParserUtil.parseIndex(userInput);
-            return new DeleteRoomCommand(index);
+            index = ParserUtil.parseIndex(userInput);
         } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteRoomCommand.MESSAGE_USAGE), pe);
+            throw pe;
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java:25]: Maybe just don't catch it?

**Review by stein414 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java (hunk 1 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java]: Updated

File: src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java (hunk 2 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java]: Updated

**Review by stein414 — APPROVED** [review]
LGTM

**Pull request merged**